share/new_eds/eds.go

```diff
@@ -0,0 +1,29 @@
+package eds
+
+import (
+	"context"
+	"io"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+// EDS is an interface for accessing extended data square data.
+type EDS interface {
+	io.Closer
```

Comment:
Do we want to enforce closing for all implementations? Generally I am fine with that, but we may also consider enforcing that on the storage/network layer only.